src/main/java/ch/njol/skript/util/WeatherType.java

```diff
@@ -105,28 +103,53 @@ public static WeatherType fromEvent(final ThunderChangeEvent e) {
 			return RAIN;
 		return CLEAR;
 	}
-	
+
+	public static WeatherType fromPlayer(final Player player) {
+		if (player.getPlayerWeather() == null) {
+			return CLEAR;
+		}
+		switch (player.getPlayerWeather()) {
+			case DOWNFALL:
+				return RAIN;
+			case CLEAR:
+				return CLEAR;
+			default:
+				throw new RuntimeException("Unknown enum value: " + player.getPlayerWeather().name());

```

Comment:
Actually I probably could have just ommited that and it would error by default right? ~~My poor Java..~~